plugins/Observability/src/test/report-generator-for-regression-testing.ssc

```diff
@@ -0,0 +1,63 @@
+// Name: Observability Plugin Report Generator For Regression Testing
+// License: Public Domain
+// Author: colossatr0n
+// Version: 1.0
+// Description: Generates reports to be used for regression testing. Run this script in two different versions of Stellarium and compare the output using some kind of differ (vim, git, diff). For macOS, file is output to "~/Library/Application Support/Stellarium/".
+
+const objectNames = [
+    "Sun",      // Sun
+    "Moon",     // Moon
+    "Jupiter",  // Solar system planet
+    "Achernar", // Interstellar object
+]
+
+// Locations from different continents
+const observerLocations = [
+    "Boise, Northern America",
+    "Brasilia, Southern America",
+    "Palestine, Northern America",
+    "Barnaul, Northern Asia",
+    "Albany, Australasia",
+    "Prague, Eastern Europe"
+]
+
+Observability.showReport(true)
+
+core.resetOutput()
+var reports = []
+for (i=0; i < observerLocations.length; i++) {
+    var location = observerLocations[i]
+    core.setObserverLocation(location, 0.0)
+    var actualLocation = core.getObserverLocation()
+    var attempts = 0
+    while (location != actualLocation) {
+        if (attempts > 100) {
+            core.debug("Couldn't update location. It appears that the location ID '" + location + "' is incorrect or the attempted location change happened too slowly.")
+            break
+        }
+        actualLocation = core.getObserverLocation()
+        attempts++
+    }
+    for (j=0; j < objectNames.length; j++) {
+        var objectName = objectNames[j]
+        core.selectObjectByName(objectName, false)
+        // Need to wait a sufficient amount of time to allow the object selection to take place.
+        // Could use core.getSelectedObjectInfo to verify the selection was successful, but using that method crashes Stellarium.
```

Comment:
Huh? Does it also crash if you wait? This should not crash, of course.

Reply:
Yeah, it crashes either way. It has something to do with the object being selected. If I go to the script console and paste:
```
core.selectObjectByName("Achernar")
core.getSelectedObjectInfo()
```
It crashes the program for my feature branch. I tried it on a released build but didn't have any issues. The error is 

```
ASSERT: "0" in file /Users/username/code/stellarium/src/core/StelObject.hpp, line 277
```
The line it's complaining about is here: https://github.com/Stellarium/stellarium/blob/51db4ee66f176d5c94f6ebe95eb0a72d33dc9306/src/core/StelObject.hpp#L277

If I select a different object, say Jupiter, I don't get the error.

Reply:
Debug mode?

Reply:
Yeah. This happened in debug mode.

Reply:
The Q_ASSERT must be removed! StelObject should deliver zero as return value of the base class.

Reply:
Fixed in master